src/ui.rs

```diff
@@ -56,18 +56,35 @@ pub fn draw<B: Backend>(f: &mut Frame<B>, app: &mut App) {
 
 fn draw_chat<B: Backend>(f: &mut Frame<B>, app: &mut App, area: Rect) {
     let text_width = area.width.saturating_sub(2) as usize;
-    let lines = app
-        .data
-        .input
-        .chars()
-        .enumerate()
-        .fold(Vec::new(), |mut lines, (idx, c)| {
-            if idx % text_width == 0 {
-                lines.push(String::new())
-            }
-            lines.last_mut().unwrap().push(c);
-            lines
-        });
+    let lines: Vec<String> =
+        app.data
+            .input
+            .chars()
+            .enumerate()
+            .fold(Vec::new(), |mut lines, (idx, c)| {
+                if idx % text_width == 0 {
+                    lines.push(String::new());
+                }
+                match c {
```

Comment:
You could combine this match with the `if` above: `if idx % text_width == 0 || c == '\n'`.

Reply:
that's subtly different, in the case that `'\n'` is input at the end of a line that will push a new string and then draw the newline character, which will mess up the cursor positioning and rendering in weird ways